### DISCUSSION TITLE
[beginner] Question about cybersecurity end2end example

### DISCUSSION THREAD

**Discussion by ArthurEly** [Xilinx/finn#953] [Q&A] [answered]
Hi!

I'm new to FINN and i started to looking at [this example](https://github.com/Xilinx/finn/tree/dev/notebooks/end2end_example/cybersecurity). I've already generated the Vivado project and opened it without errors.

I have two main questions:

1. how can i test the generated IP? I want to infer the neural network on a Nexys 4 (xc7a100tcsg324-1)
2. the example indicates that the input of the onnx model is a 600-wide vector, but in the generated ip, the input is 40 bits wide. Is this correct?

Below are pictures of the generated ip in Vivado
![image](https://github.com/Xilinx/finn/assets/58867196/172073d3-9ee0-4429-9b62-20fd7ba0eb40)
![image](https://github.com/Xilinx/finn/assets/58867196/ad7de3ba-c111-4f62-9f8c-3ddb061cf854)

and the onnx of the generated ip
![image](https://github.com/Xilinx/finn/assets/58867196/a8869ed5-258f-4757-97d2-5823082b4751)

Thank you!

**Comment by fpjentzsch** [comment, marked as answer]
Hi,

1) Since this is an FPGA development board only (no ARM CPU or PYNQ support), you will have to integrate the "stitched_ip" generated by FINN into your own block design that streams data in/out of the accelerator via the AXI-Stream interfaces. Maybe you could use a softcore processor to feed the data from memory/storage.

2) The input width of 40 is correct because it is determined by the folding (parallelism) of the first MVAU layer, more specifically the SIMD folding parameter. You could adjust the folding to process more or fewer bits of the 600 bit input in each FPGA clock cycle.

**Reply by ArthurEly** [reply]
Hi, Felix
Thank you very much!   :)
